packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -218,6 +218,104 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
      *          RELAYER FUNCTIONS         *
      **************************************/
 
+    /**
+     * @notice Called by Relayer to execute a slow + fast relay from L2 to L1, fulfilling a corresponding deposit order.
+     * @dev There can only be one pending relay for a deposit. This method is effectively the relayDeposit and
+     * speedUpRelay methods concatenated. This could potentially be refactored to just call each method, but there
+     * are some gas savings in combining the transfers and hash computations.
+     * @dev Caller must have approved this contract to spend the total bond + amount - fees for `l1Token`.
+     * @param depositData the deposit data struct containing all the user's deposit information.
+     * @param realizedLpFeePct LP fee calculated off-chain considering the L1 pool liquidity at deposit time, before
+     *      quoteTimestamp. The OO acts to verify the correctness of this realized fee. Can not exceed 50%.
+     */
+    function relayAndSpeedUp(DepositData memory depositData, uint64 realizedLpFeePct) public nonReentrant() {
```

Comment:
DO you think we could also change the `relayDeposit` inputs to take in `DepositData memory depositData` instead of the destructured params?

Reply:
We can -- just didn't want to modify more than necessary. I did it with the new method bc of stack too deep errors.